src/memcached.c

```diff
@@ -285,6 +297,47 @@ static void submit_gauge2(const char *type, const char *type_inst,
   plugin_dispatch_values(&vl);
 }
 
+static gauge_t calculate_rate(gauge_t part, gauge_t total, gauge_t *prev_part,
```

Comment:
I find it a bit unintuitive that you're effectively calculating the rated of `gauge_t`. It looks like the raw metrics are integers, can you change these functions to take `derive_t` instead?

Reply:
The current code has following variables:
```
  gauge_t hits = NAN;
  gauge_t gets = NAN;
  gauge_t incr_hits = NAN;
```
https://github.com/collectd/collectd/pull/2385/commits/e8fdf93c32d752022478bed93315950f499bc271#diff-e842cfa6e2070dbfa4c72d77b5b2fc71R351

These variables are passed to `calculate_ratio()` and `calculate_ratio2()` functions, so they take `gauge_t` type as  as argument. What to do?

Reply:
Convert those variables to `derive_t`, too. I'm fairly certain they were only `gauge_t` so `rate = 100.0 * hits / gets;` could be calculated without a cast.

Reply:
Updated, variables converted to `derive_t`.